docs/intro/contributing.txt

```diff
@@ -517,6 +517,27 @@ When you're done previewing the patch, hit the ``q`` key to return to the
 command line. If the patch's content looked okay, it's time to commit the
 changes.
 
+Pre-commit checks
```

Comment:
I'm inlined to say this belongs in internals/contributing, perhaps writing-code/coding-style. I think it's more of an advanced thing that we don't need to flag for contributors just trying to learn the basics.

Reply:
I agree it could fit there as well. This section is fairly long as it is. 

My thinking for suggesting here is that experienced developers are more likely to know about pre-commit and decide if they want to use these? It's new folk that probably need a bit more guidance?

Reply:
Others should give their opinion but I'd expect the hooks are most useful for regular contributors, which likely ends up as a very small percentage of those working through the tutorial.

Reply:
I agree with Tim here. Let's not overwhelm newcomers. The pre-commit hooks will run on CI anyway.

Reply:
I'm glad you agree as I started to rewrite the patch to do this yesterday. If we're aiming it at more regular contributors then I'm more inclined to have the pre-commit hooks pointing at the tox envs. This then avoids the concerns expressed about needing to keep bumping the versions?